apps/update-masterkeys.nix

```
@@ -0,0 +1,54 @@
+{ pkgs, ... }@inputs:
+let
+  inherit (import ../nix/lib.nix inputs)
+    ageMasterEncrypt
+    ageMasterDecrypt
+    validRelativeSecretPaths
+    ;
+in
+pkgs.writeShellScriptBin "agenix-update-masterkeys" ''
+  set -uo pipefail
+
+  function die() { echo "[1;31merror:[m $*" >&2; exit 1; }
+  function show_help() {
+    echo 'Usage: agenix update-masterkeys'
+    echo 'Update all stored secrets with a new set of masterkeys.'
+  }
+
+  if [[ ! -e flake.nix ]] ; then
+    die "Please execute this script from your flake's root directory."
+  fi
+
+  function cleanup() {
+    [[ -e "$CLEARTEXT_FILE" ]] && rm "$CLEARTEXT_FILE"
+    [[ -e "$ENCRYPTED_FILE" ]] && rm "$ENCRYPTED_FILE"
+  }; trap "cleanup" EXIT
+
+  ${builtins.concatStringsSep "" (
+    builtins.map (
+      path: # bash
+      ''
+        CLEARTEXT_FILE=$(mktemp)
+        ENCRYPTED_FILE=$(mktemp)
+
+        shasum_before="$(sha512sum "${path}")"}
```

```suggestion
        shasum_before="$(sha512sum "${path}")"
```

In think this may cause the script to always reencrypt each file in its current form

---

Ahhhhh that makes sense, sorry for missing it 😄